implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -0,0 +1,261 @@
+        // TODO
+        return getMetricBuilder(meter.getId()).build();
+    }
+
+    Metric writeGauge(Gauge gauge) {
+        return getMetricBuilder(gauge.getId())
+                .setGauge(io.opentelemetry.proto.metrics.v1.Gauge.newBuilder()
+                        .addDataPoints(NumberDataPoint.newBuilder()
+                                .setTimeUnixNano(TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime()))
+                                .setAsDouble(gauge.value())
+                                .addAllAttributes(getTagsForId(gauge.getId()))
+                                .build()))
+                .build();
+    }
+
+    Metric writeCounter(Counter counter) {
+        StartTimeAwareCumulativeCounter otlpCounter = (StartTimeAwareCumulativeCounter) counter;
+        return getMetricBuilder(otlpCounter.getId())
+                .setSum(Sum.newBuilder()
+                        .addDataPoints(NumberDataPoint.newBuilder()
+                                .setStartTimeUnixNano(otlpCounter.getStartTimeNanos())
+                                .setTimeUnixNano(TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime()))
+                                .setAsDouble(counter.count())
+                                .addAllAttributes(getTagsForId(counter.getId()))
+                                .build())
+                        .setIsMonotonic(true)
+                        .setAggregationTemporality(AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE)
+                        .build())
+                .build();
+    }
+
+    private Metric writeHistogramSupport(HistogramSupport histogramSupport) {
+        Metric.Builder metricBuilder = getMetricBuilder(histogramSupport.getId());
+        boolean isTimeBased = histogramSupport instanceof Timer;
+        HistogramSnapshot histogramSnapshot = histogramSupport.takeSnapshot();
+        long wallTimeNanos = TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime());
+
+        HistogramDataPoint.Builder histogramDataPoint = HistogramDataPoint.newBuilder()
+                .addAllAttributes(getTagsForId(histogramSupport.getId()))
+                // TODO start time
+                .setTimeUnixNano(wallTimeNanos)
+                .setSum(isTimeBased ? histogramSnapshot.total(getBaseTimeUnit()) : histogramSnapshot.total())
+                .setCount(histogramSnapshot.count());
+
+        // if histogram enabled, add histogram buckets
+        if (histogramSnapshot.histogramCounts().length != 0) {
+            for (CountAtBucket countAtBucket : histogramSnapshot.histogramCounts()) {
+                histogramDataPoint.addExplicitBounds(isTimeBased ? countAtBucket.bucket(getBaseTimeUnit()) : countAtBucket.bucket());
+                histogramDataPoint.addBucketCounts((long) countAtBucket.count());
+            }
+            metricBuilder.setHistogram(Histogram.newBuilder()
+                    .setAggregationTemporality(AggregationTemporality.AGGREGATION_TEMPORALITY_DELTA)
```

Comment:
I'm testing out sending histogram data with buckets to an Otel Collector that exports Prometheus, and I don't see a histogram exported in the Prometheus endpoint with delta temporality. Also, bucket counts should not add the counts in lower buckets like Prometheus histograms.